packages/studioCMS/src/utils/renderers/markdoc.ts

```diff
@@ -1,11 +1,23 @@
 import Markdoc from '@markdoc/markdoc';
+import Config from 'virtual:studiocms/config';
+import React, { type ReactNode } from 'react';
 
-export async function renderMarkDoc(input: string): Promise<string> {
-	const ast = Markdoc.parse(input);
+const {
+	markdocConfig: { argParse, renderType, transformConfig },
+} = Config;
 
-	const content = Markdoc.transform(ast);
+export async function renderMarkDoc(input: string): Promise<string | ReactNode> {
+	const ast = Markdoc.parse(input, argParse);
+	const content = Markdoc.transform(ast, transformConfig);
 
-	const html = Markdoc.renderers.html(content);
-
-	return html;
+	switch (renderType) {
+		case 'react':
+			return Markdoc.renderers.react(content, React);
+		case 'react-static':
+			return Markdoc.renderers.reactStatic(content);
```

Comment:
react => return ReactNode
react-static => return string so html

Reply:
for the ReactNode, it might be work working out a implementation with the [`Astro ContainerAPI`](https://docs.astro.build/en/reference/container-reference/)

Reply:
> for the ReactNode, it might be work working out a implementation with the [`Astro ContainerAPI`](https://docs.astro.build/en/reference/container-reference/)

yes, but it’s in an experimental phase, so now we can remove the react option and add it later

Reply:
This whole project(studioCMS) is in experimental state as well (and we are already using the container api :stuck_out_tongue: )

Reply:
> This whole project(studioCMS) is in experimental state as well (and we are already using the container api 😛 )

yeah, so we can use container api...

Reply:
indeed!  Hakuna matata!

I like living on the edge.... we are using ALOT of experimental/beta stuff since astroDB is experimental as well